### PR TITLE
fix: settings UI overhaul

### PR DESCRIPTION
**Pull request by tiankii**
## Summary
- Built a custom Switch to standardize toggle visuals across iOS/Android, then wired it into Notifications and Security settings.
- Brought Settings screens into a single layout system with SettingsGroup: aligned dividers, padding, and row spacing across Notifications and API screens.
- Tightened SettingsRow behavior: right-action spacing, touch-area-safe truncation, and consistent text clamping for long labels.
- Aligned Account ID and Access Methods rows to the same geometry as other settings rows (icon sizing and spacing).
- Refined row content formats, including Receive currency single-line labels that match the settings pattern.
- Updated the POS icon asset to match the new icon set.
- Cleaned up Phone LN address logic: renamed the component, removed unused subtitle handling, only show when a verified phone exists, and reuse login-method data to avoid duplicate queries.
- Added tests to cover phone LN visibility and settings row truncation edge cases.

### CODE REVIEW

File: app/screens/settings-screen/settings/phone-ln-address.tsx [hunk 1]
```diff
@@ -1,63 +1,48 @@
...
+  const { phoneVerified } = useLoginMethods()
   const { LL } = useI18nContext()
 
-  const hasUsername = Boolean(data?.me?.phone)
-  const lnAddress = `${data?.me?.phone}@${hostName}`
+  const phoneNumber = data?.me?.phone
+  const isPhoneVerified = phoneVerified && Boolean(phoneNumber)
+  const lnAddress = `${phoneNumber}@${hostName}`
+
+  if (!isPhoneVerified) return null
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/settings-screen/settings/phone-ln-address.tsx]: Done!

```diff
diff --git a/app/screens/settings-screen/settings/phone-ln-address.tsx b/app/screens/settings-screen/settings/phone-ln-address.tsx
index e800a5605..431119891 100644
--- a/app/screens/settings-screen/settings/phone-ln-address.tsx
+++ b/app/screens/settings-screen/settings/phone-ln-address.tsx
@@ -3,7 +3,6 @@ import Clipboard from "@react-native-clipboard/clipboard"
 import { useTheme } from "@rn-vui/themed"
 
 import { GaloyIcon } from "@app/components/atomic/galoy-icon"
-import { useSettingsScreenQuery } from "@app/graphql/generated"
 import { useI18nContext } from "@app/i18n/i18n-react"
 import { useLoginMethods } from "@app/screens/settings-screen/account/login-methods-hook"
 import { useAppConfig } from "@app/hooks"
@@ -18,15 +17,12 @@ export const PhoneLnAddress: React.FC = () => {
   } = useTheme()
   const hostName = appConfig.galoyInstance.lnAddressHostname
 
-  const { data, loading } = useSettingsScreenQuery()
-  const { phoneVerified } = useLoginMethods()
+  const { loading, phone, phoneVerified } = useLoginMethods()
   const { LL } = useI18nContext()
 
-  const phoneNumber = data?.me?.phone
-  const isPhoneVerified = phoneVerified && Boolean(phoneNumber)
-  const lnAddress = `${phoneNumber}@${hostName}`
+  if (!phoneVerified || !phone) return null
 
-  if (!isPhoneVerified) return null
+  const lnAddress = `${phone}@${hostName}`
```

File: app/components/atomic/switch/switch.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  const animatedThumbStyle = useAnimatedStyle(() => {
+    const translateX = progress.value * (TRACK_WIDTH - THUMB_SIZE - THUMB_OFFSET * 2)
+    return {
+      transform: [{ translateX }],
+    }
+  })
+
+  return (
+    <Pressable onPress={handlePress} disabled={disabled} style={style} testID={testID}>
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/atomic/switch/switch.tsx]: Done!

File: __tests__/screens/settings-screen/settings-screen.spec.tsx [hunk 3]
```diff
@@ -420,4 +422,45 @@ describe("Settings Screen", () => {
...
+    await act(
+      () =>
+        new Promise((resolve) => {
+          setTimeout(resolve, 10)
+        }),
+    )
+
+    expect(screen.queryByText("Set your lightning address")).toBeNull()
+    expect(screen.queryByText("+50365055539@blink.sv")).toBeNull()
+  })
```
[the remark this answers is not in the document]
Comment (grimen) [reply __tests__/screens/settings-screen/settings-screen.spec.tsx:466]: @esaugomez31 I agree with Copilot here, and should be simple addition.

File: app/components/atomic/switch/switch.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+  const progress = useSharedValue(value ? 1 : 0)
+
+  React.useEffect(() => {
+    progress.value = withTiming(value ? 1 : 0, { duration: ANIMATION_DURATION })
+  }, [value, progress])
+
+  const handlePress = () => {
+    if (!disabled) {
+      progress.value = withTiming(value ? 0 : 1, { duration: ANIMATION_DURATION })
```
[the remark this answers is not in the document]
Comment (grimen) [reply app/components/atomic/switch/switch.tsx:57]: @esaugomez31 Please verify?

File: app/screens/settings-screen/account/id.tsx [hunk 5]
```diff
@@ -28,72 +35,15 @@ export const AccountId: React.FC = () => {
...
-    </View>
+  const AccountIdRow = () => (
+    <SettingsRow
+      loading={loading}
+      title={maskedAccountId}
+      action={null}
+      rightIcon={<GaloyIcon name="copy-paste" size={24} color={colors.primary} />}
+      rightIconAction={copyToClipboard}
+    />
   )
```
[the remark this answers is not in the document]
Comment (grimen) [reply app/screens/settings-screen/account/id.tsx:46]: @esaugomez31 Please verify?

File: app/screens/settings-screen/api-screen.tsx [hunk 6]
```diff
@@ -45,65 +77,28 @@ export const ApiScreen: React.FC = () => {
     Linking.openURL(item.link)
   }
 
+  const apiSettings = apiItems.map((item) => {
+    const ApiItemRowWithItem: React.FC = () => (
+      <ApiItemRow colors={colors} item={item} onPress={handleItemPress} styles={styles} />
+    )
+    ApiItemRowWithItem.displayName = `ApiItemRow-${item.id}`
+    return ApiItemRowWithItem
+  })
```
[the remark this answers is not in the document]
Comment (grimen) [reply app/screens/settings-screen/api-screen.tsx:86]: @esaugomez31 Please verify?

File: app/screens/settings-screen/notifications-screen.tsx [hunk 7]
```diff
@@ -200,47 +198,47 @@ export const NotificationSettingsScreen: React.FC = () => {
...
-        </ListItem>
-        {!isLast && <Divider color={colors.grey4} />}
-      </React.Fragment>
+  const pushNotificationSettings = categoriesArray.map((category) => {
+    const NotificationRowWithCategory: React.FC = () => (
+      <NotificationRow category={category} />
     )
+    NotificationRowWithCategory.displayName = `NotificationRow-${category}`
+    return NotificationRowWithCategory
   })
```
[the remark this answers is not in the document]
Comment (grimen) [reply app/screens/settings-screen/notifications-screen.tsx:229]: @esaugomez31 Please verify?

File: app/components/atomic/switch/switch.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/atomic/switch/switch.tsx:57]: Ok I kept a single animation per toggle by skipping the next useEffect animation when the press already animates, so there’s no double‑animation. I also switched the handler to onPressIn and added hitSlop to avoid missed taps on quick pressesI kept a single animation per toggle by skipping the next useEffect animation when the press already animates, so there’s no double‑animation. I also switched the handler to onPressIn and added hitSlop to avoid missed taps on quick presses

File: app/screens/settings-screen/account/id.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/settings-screen/account/id.tsx:46]: Done!

Added a `displayName` to the inline `AccountIdRow` component to satisfy the readability/debugging concern without changing the layout

File: app/screens/settings-screen/api-screen.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/settings-screen/api-screen.tsx:86]: We can’t pass `ApiItemRow` elements directly because `SettingsGroup` only accepts `React.FC[]`. To keep `SettingsGroup` unchanged, we kept the wrapper and added a `displayName` per item for clarity/debugging. Given that `SettingsGroup` expects functions, this extra nesting is minimal and not a performance concern here

File: app/screens/settings-screen/notifications-screen.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/settings-screen/notifications-screen.tsx:229]: For this one, same situation as the API rows: SettingsGroup only accepts React.FC[], so each category needs a wrapper. We kept the wrapper, added displayName, and NotificationRow already accepts category directly. No extra work needed unless we change SettingsGroup

File: __tests__/screens/settings-screen/settings-screen.spec.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (esaugomez31) [reply __tests__/screens/settings-screen/settings-screen.spec.tsx:466]: Ok, I Added tests for long titles and subtitles, asserting numberOfLines=1 and ellipsizeMode="tail" on SettingsRow
